### PR TITLE
Allow Encoded.InCharset to be specified for PropertiesGenerator

### PR DESCRIPTION
**Pull request by brentvukmer**
For #152 .. happy to use `configure` method per your feedback, but wanted to push what I had to see what you think.

### CODE REVIEW

File: generators/src/main/java/com/pholser/junit/quickcheck/generator/java/util/PropertiesGenerator.java [hunk 1]
```diff
@@ -25,47 +25,70 @@ a copy of this software and associated documentation files (the
...
         return properties;
     }
 
     @SuppressWarnings("unchecked")
-    @Override public boolean canRegisterAsType(Class<?> type) {
+    @Override
+    public boolean canRegisterAsType(Class<?> type) {
         Set<Class<?>> exclusions =
-            new HashSet<>(asList(Object.class, Hashtable.class, Map.class, Dictionary.class));
+                new HashSet<>(asList(Object.class, Hashtable.class, Map.class, Dictionary.class));
```
Comment (brentvukmer) [inline generators/src/main/java/com/pholser/junit/quickcheck/generator/java/util/PropertiesGenerator.java]: Is my IntelliJ-default formatting irritating?

File: generators/src/main/java/com/pholser/junit/quickcheck/generator/java/util/PropertiesGenerator.java [hunk 2]
```diff
@@ -25,47 +25,70 @@ a copy of this software and associated documentation files (the
...
+    /**
+     * If string generation for property keys and values should limited to a given {@link Encoded.InCharset},
+     * then use this constructor.
+     */
+    public PropertiesGenerator(Encoded.InCharset charset) {
+        this(new Encoded());
+        ((Encoded) stringGenerator).configure(charset);
     }
 
-    @Override public Properties generate(SourceOfRandomness random, GenerationStatus status) {
```
Comment (brentvukmer) [inline generators/src/main/java/com/pholser/junit/quickcheck/generator/java/util/PropertiesGenerator.java]: IntelliJ default formatting.  I can back it out if you want.

File: generators/src/main/java/com/pholser/junit/quickcheck/generator/java/util/PropertiesGenerator.java [hunk 3]
```diff
@@ -25,47 +25,70 @@ a copy of this software and associated documentation files (the
...
+    @Override
+    public Properties generate(SourceOfRandomness random, GenerationStatus status) {
         int size = status.size();
 
         Properties properties = new Properties();
         for (int i = 0; i < size; ++i) {
             properties.setProperty(
-                stringGenerator.generate(random, status),
-                stringGenerator.generate(random, status));
+                    stringGenerator.generate(random, status),
```
Comment (brentvukmer) [inline generators/src/main/java/com/pholser/junit/quickcheck/generator/java/util/PropertiesGenerator.java]: More IntelliJ default formatting.

Comment (pholser) [inline generators/src/main/java/com/pholser/junit/quickcheck/generator/java/util/PropertiesGenerator.java]: @brentvukmer Yeah, I'd prefer keeping both scope indent and long-line indent at four spaces. Thanks!

**Comment by pholser** [comment]
@brentvukmer Using a constructor that accepts either the `AbstractStringGenerator` or an `InCharset` annotation won't do what you're wanting.

Imagine a property method such as:

    @Property public void x(@Encoded.InCharset("US-ASCII") Properties p) {
        // ...
    }

To generate values for parameter `p`, junit-quickcheck will look for generators from the `ServiceLoader` for `com.pholser.junit.quickcheck.generator.Generator` that are type-compatible with `java.util.Properties` and honor all of the configuration annotations on `p`. To honor a configuration annotation, a generator provides a `public` method named `configure` that has one parameter, whose type is a configuration annotation type.

If `PropertiesGenerator` doesn't have a method:

    public void configure(Encoded.InCharset c) {
    }

there will be no opportunity for junit-quickcheck to pass the configuration annotation on to `PropertiesGenerator` so that it can select an appropriate `StringGenerator` with which to generate keys and values.

See [Configuring Generators](http://pholser.github.io/junit-quickcheck/site/0.8-alpha-4/usage/configuring.html) for more details.

**Comment by brentvukmer** [comment]
Thank you for the feedback!   I appreciate the time spent on educating me, for something you could have done so quickly yourself.

**Comment by pholser** [comment]
@brentvukmer One more thing - would you squash your commits into a single commit, and force-push your branch? That way, the commit history will look smoother, without the fiddling with formatting, etc. This amounts to, I believe (on your feature branch):

    $ git rebase -i HEAD~3
    $ git push -f origin <your-branch>

**Comment by brentvukmer** [comment]
@pholser yes - thanks for your patience, should have done that already.

**Pull request merged**